File: etablissement/src/main/java/com/projetfinal/etablissement/entity/Cours.java
package com.projetfinal.etablissement.entity;

import java.time.LocalTime;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
public class Cours {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	@JsonView(Vue.Common.class)
	private Integer id;
	@JsonView(Vue.Common.class)
	private LocalTime heureDebut;
	@JsonView(Vue.Common.class)
	private LocalTime heureFin;
	@JsonView(Vue.Common.class)
	private int day; // 0 à 6, pour être utilisé avec les méthodes de java.util.Date.getDay()
	@ManyToOne
	@JsonView(Vue.Common.class)
	private Professeur professeur;
	@ManyToOne
	@JsonView(Vue.Common.class)
	private Matiere matiere;
	@ManyToOne
	@JsonView(Vue.Common.class)
	private SalleClasse salle;
	
	public Cours() {
	}

	public Cours(LocalTime heureDebut, LocalTime heureFin, Professeur professeur, Matiere matiere,
			SalleClasse salle, int day) {
		super();
		this.heureDebut = heureDebut;
		this.heureFin = heureFin;
		this.professeur = professeur;
		this.matiere = matiere;
		this.salle = salle;
		this.day = day;
	}

	public Integer getId() {
		return id;
	}

	public LocalTime getHeureDebut() {
		return heureDebut;
	}

	public void setHeureDebut(LocalTime heureDebut) {
		this.heureDebut = heureDebut;
	}

	public LocalTime getHeureFin() {
		return heureFin;
	}

	public void setHeureFin(LocalTime heureFin) {
		this.heureFin = heureFin;
	}

	public Professeur getProfesseur() {
		return professeur;
	}

	public void setProfesseur(Professeur professeur) {
		this.professeur = professeur;
	}

	public Matiere getMatiere() {
		return matiere;
	}

	public void setMatiere(Matiere matiere) {
		this.matiere = matiere;
	}

	public SalleClasse getSalle() {
		return salle;
	}

	public void setSalle(SalleClasse salle) {
		this.salle = salle;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Cours other = (Cours) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}

	public int getDay() {
		return day;
	}

	public void setDay(int day) {
		this.day = day;
	}
	
	
}

File: etablissementAngular/src/app/model/cours.ts
import { Matiere } from "./matiere";
import { Professeur } from "./professeur";
import { SalleClasse } from "./salle-classe";

export class Cours {

private _id: number;
private _heureDebut: string;
private _heureFin: string;
private _professeur: Professeur;
private _matiere: Matiere
private _salle: SalleClasse;


	constructor(id: number, heureDebut: string , heureFin: string, professeur: Professeur, matiere: Matiere, salle: SalleClasse) {
		this._id = id;
		
		this._heureDebut = heureDebut;
		this._heureFin = heureFin;
		this._professeur = professeur;
		this._matiere = matiere;
		this._salle = salle;
	}


    /**
     * Getter id
     * @return {number}
     */
	public get id(): number {
		return this._id;
	}

    /**
     * Getter heureDebut
     * @return {string}
     */
	public get heureDebut(): string {
		return this._heureDebut;
	}

    /**
     * Getter heureFin
     * @return {string}
     */
	public get heureFin(): string {
		return this._heureFin;
	}

    /**
     * Getter professeur
     * @return {Professeur}
     */
	public get professeur(): Professeur {
		return this._professeur;
	}

    /**
     * Getter matiere
     * @return {Matiere}
     */
	public get matiere(): Matiere {
		return this._matiere;
	}

    /**
     * Getter salle
     * @return {SalleClasse}
     */
	public get salle(): SalleClasse {
		return this._salle;
	}

    /**
     * Setter id
     * @param {number} value
     */
	public set id(value: number) {
		this._id = value;
	}

    /**
     * Setter heureDebut
     * @param {string} value
     */
	public set heureDebut(value: string) {
		this._heureDebut = value;
	}

    /**
     * Setter heureFin
     * @param {string} value
     */
	public set heureFin(value: string) {
		this._heureFin = value;
	}

    /**
     * Setter professeur
     * @param {Professeur} value
     */
	public set professeur(value: Professeur) {
		this._professeur = value;
	}

    /**
     * Setter matiere
     * @param {Matiere} value
     */
	public set matiere(value: Matiere) {
		this._matiere = value;
	}

    /**
     * Setter salle
     * @param {SalleClasse} value
     */
	public set salle(value: SalleClasse) {
		this._salle = value;
	}


}
